rest/route/commit_queue.go

```diff
@@ -124,3 +125,46 @@ func (cq *commitQueueClearAllHandler) Run(ctx context.Context) gimlet.Responder
 		ClearedCount int `json:"cleared_count"`
 	}{clearedCount})
 }
+
+type commitQueueEnqueueItemHandler struct {
+	item string
+
+	sc data.Connector
+}
+
+func makeCommitQueueEnqueueItem(sc data.Connector) gimlet.RouteHandler {
+	return &commitQueueEnqueueItemHandler{
+		sc: sc,
+	}
+}
+
+func (cq commitQueueEnqueueItemHandler) Factory() gimlet.RouteHandler {
+	return &commitQueueEnqueueItemHandler{
+		sc: cq.sc,
+	}
+}
+
+func (cq *commitQueueEnqueueItemHandler) Parse(ctx context.Context, r *http.Request) error {
+	vars := gimlet.GetVars(r)
+	cq.item = vars["item"]
+
+	return nil
+}
+
+func (cq *commitQueueEnqueueItemHandler) Run(ctx context.Context) gimlet.Responder {
+	patch, err := cq.sc.FindPatchById(cq.item)
+	if err != nil {
+		if _, ok := err.(gimlet.ErrorResponse); ok {
+			return gimlet.MakeJSONErrorResponder(err)
+		}
+		return gimlet.MakeJSONInternalErrorResponder(errors.Wrap(err, "can't find item"))
```

Comment:
you don't need to type check for error responses, the Make*ErrorResponder functions wrap and unwrap as appropriate.

Reply:
neat.
Done.